module-apps/apps-common/ApplicationCommon.cpp

```diff
@@ -772,243 +801,113 @@ namespace app
-        else if (id == ID::PhoneLock) {
-            switchWindow(
-                gui::popup::resolveWindowName(id), gui::ShowMode::GUI_SHOW_INIT, nullptr, SwitchReason::PhoneLock);
-        }
-        else if (id == ID::PhoneLockInput || id == ID::PhoneLockChangeInfo) {
-            auto popupParams = static_cast<const gui::PhoneUnlockInputRequestParams *>(params);
-
-            switchWindow(
-                gui::popup::resolveWindowName(id),
-                std::make_unique<locks::LockData>(popupParams->getLock(), popupParams->getPhoneLockInputTypeAction()));
-        }
-        else if (id == ID::SimLock || id == ID::SimInfo) {
-            auto popupParams = static_cast<const gui::SimUnlockInputRequestParams *>(params);
-
-            switchWindow(gui::popup::resolveWindowName(id),
-                         std::make_unique<locks::SimLockData>(popupParams->getLock(),
-                                                              popupParams->getSimInputTypeAction(),
-                                                              popupParams->getErrorCode()));
-        }
-        else if (id == ID::Alarm) {
-            auto popupParams =
-                const_cast<gui::AlarmPopupRequestParams *>(static_cast<const gui::AlarmPopupRequestParams *>(params));
-            switchWindow(gui::popup::resolveWindowName(id),
-                         std::make_unique<gui::AlarmPopupRequestParams>(popupParams));
-        }
-        else {
-            switchWindow(gui::popup::resolveWindowName(id));
+        auto request = windowsPopupQueue->popRequest(getPopupFilter());
+        if (request) {
+            auto popup = std::string(magic_enum::enum_name(request->getPopupParams().getPopupId()));
+            LOG_DEBUG("handling popup: %s", popup.c_str());
+            /// request handle actually switches window to popup window
+            auto retval = request->handle();
+            if (not retval) {
+                LOG_ERROR("Popup %s handling failure, please check registered blueprint!", popup.c_str());
+            }
+            return retval;
+            LOG_ERROR("no blueprint for popup: %s", popup.c_str());
         }
+        return false;
     }
 
+    // oh this is kind of bad... we require popup start window to properly abort selected popup XD
     void ApplicationCommon::abortPopup(gui::popup::ID id)
     {
         const auto popupName = gui::popup::resolveWindowName(id);
-
-        if (getCurrentWindow()->getName() == popupName) {
-            returnToPreviousWindow();
-        }
-        else {
-            popWindow(popupName);
-        }
+        LOG_INFO("abort popup: %s from window %s", popupName.c_str(), getCurrentWindow()->getName().c_str());
+        windowsStack().popTo(popupName);
+        returnToPreviousWindow();
     }
 
-    bool ApplicationCommon::isPopupPermitted([[maybe_unused]] gui::popup::ID popupId) const
+    bool ApplicationCommon::handleUI_DBNotification(sys::Message *msg, const UiNotificationFilter &filter)
```

Comment:
fix that name style ? we do not use "_" in function names.

Reply:
Then we would have UIDB <- two acronyms glued in. or do you want to have `handeUserInterfaceDatabaseNotification` ? :)

Reply:
talked: `userInterfaceDBNotification` is a final name